.github/ISSUE_TEMPLATE/25_api_proposal.md

```diff
@@ -0,0 +1,53 @@
+---
+name: API proposal
+about: Propose a change to the public API surface
+title: ''
+labels: api-suggestion
+assignees: ''
+
+---
+
+## Background and Motivation
+
+<!--
+We welcome API proposals! We have a process to evaluate the value and shape of new API. There is an overview of our process [here](https://github.com/dotnet/aspnetcore/blob/main/docs/APIReviewProcess.md). This template will help us gather the information we need to start the review process.
+First, please describe the purpose and value of the new API here.
+-->
+
+## Proposed API
+
+<!--
+Please provide the specific public API signature diff that you are proposing. For example:
+```diff
+namespace System.Collections.Generic
+{
+-    public class HashSet<T> : ICollection<T>, ISet<T> {
```

Comment:
Should we use ASP.NET Core specific examples? Also what's the difference between this template and a design proposal? How is a contributor supposed to figure out what works best (because an API usually doesn't live in isolation without a design that goes with it)?

Reply:
> Should we use ASP.NET Core specific examples?
Yes

> How is a contributor supposed to figure out what works best (because an API usually doesn't live in isolation without a design that goes with it)?

This is a good point but the design proposal is heavy for adding scenario based APIs. If you can't describe it in a single issue like this then it's a good candidate for a design proposal. This is for adding APIs like:

- https://github.com/dotnet/aspnetcore/issues/5877
- https://github.com/dotnet/aspnetcore/issues/12978
- https://github.com/dotnet/aspnetcore/issues/13270
- https://github.com/dotnet/aspnetcore/issues/17972

Etc

Reply:
Can you guys please provide good examples to include then?

Reply:
@davidfowl, @pranavkm I'm waiting for relevant examples here before I merge.

Reply:
Added a suggestion